upgrade_downgrade/test_canary_upgrade_downgrade.sh

```diff
@@ -0,0 +1,193 @@
+#!/bin/bash
+
+# Copyright Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+set -o pipefail
+set -x
+
+# TODO: Remove hardcoded versions and pass it from run_upgrade_downgrade_test
+# as command-line arguments like test_upgrade_downgrade.sh
+FROM_VERSION="istio-1.7-alpha.a8c04f92e236676977b3ed58132437d9fafc9aed"
+TO_VERSION="istio-1.8-alpha.f0f221ae1024fb2d4a93b355546f094376fb2e47"
+ISTIOPATH="$GOPATH/src/istio.io"
+
+FROM_DIR="${ISTIOPATH}/tools/upgrade_downgrade/from_dir.g9xSYb/${FROM_VERSION}"
+TO_DIR="${ISTIOPATH}/tools/upgrade_downgrade/to_dir.MrCCp4/${TO_VERSION}"
+TMP_DIR="${ISTIOPATH}/tools/upgrade_downgrade/templates"
+POD_FORTIO_LOG="${TMP_DIR}/fortio.log"
+
+FROM_ISTIOCTL="${FROM_DIR}/bin/istioctl"
+TO_ISTIOCTL="${TO_DIR}/bin/istioctl"
+
+# Maximum % of 503 response that cannot exceed
+MAX_503_PCT_FOR_PASS="15"
+# Maximum % of connection refused that cannot exceed
+# Set it to high value so it fails for explicit sidecar issues
+MAX_CONNECTION_ERR_FOR_PASS="30"
+SERVICE_UNAVAILABLE_CODE="503"
+CONNECTION_ERROR_CODE="-1"
+
+
+# Install Istio 1.7 minimal profile (only istiod)
+echo "Installing Istio ${FROM_VERSION}"
+${FROM_ISTIOCTL} install -y --set profile=minimal
+kubectl wait --all --for=condition=Ready pods -n istio-system --timeout=5m
+
+# Deploy sample application
+kubectl create namespace test
```

Comment:
i think this will return error if it exists, so ignore that

Reply:
Probably I need to use something like `resetCluster` like in the other test. I'll fix it. Thanks :)